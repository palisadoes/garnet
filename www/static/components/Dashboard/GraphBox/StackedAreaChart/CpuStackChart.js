/**
 * Component: CpuStackChart
 * Purpose: Displays CpuStackChart of stacked data
 * Properties:
 *  url : local url prefix
 *  agentId : Agent Id in database
 *  stackType: Graph's stack type
 **/

 //React and React Bootstrap imports
import React, { Component } from 'react';
//HTTP Promise library import
import axios from 'axios';
//D3.js import
import * as d3 from 'd3';
//Moment.js import
import moment from 'moment';

//Recharts component imports
import { AreaChart, Area, Axis, XAxis, YAxis, CartesianGrid, Tooltip, Legend} from 'recharts';
import { ResponsiveContainer } from 'recharts';

class CpuStackChart extends Component {

  constructor(props) {
    super(props);

    this.state = {
      data: [],
    }
  }

  //Runs after Component is Loaded
  componentDidMount() {
    var _this = this;

    axios.get(this.props.url + this.props.agentId + '/' + this.props.stackType)
      .then(function (response) {
        var data = response.data;
        //Times each time by 100 to prep for conversion
        data.forEach(function(d) {
          d.timestamp = d.timestamp * 1000;
         });
         //Sets data in component
        _this.setState({
          data: response.data
        });

      })
      .catch(function (error) {
        console.log(error);
      });
  }

  //Generates graph ticks for X Axis
  getTicks(data){
  	if (!data || !data.length ) {return [];}
    const domain = [new Date(data[0].timestamp), new Date(data[data.length - 1].timstamp)];
  	const scale = d3.scaleTime().domain(domain).range([0,1]);
    const ticks = scale.ticks(d3.timeHour, 1);
    return ticks.map(entry => +entry);
  };

  //Formats unix times to hours and minutes
  dateFormat(time) {
    return moment(time).format('HH:mm');
  }

  //Formats unix times to local Month/Day/Hour/Minutes format
  toolTipDateFormat(time) {
    return moment(time).format('lll');
  }

  render(){
    return(
      <ResponsiveContainer width="100%" height="100%" aspect={2}>
        <AreaChart width={600} height={300} data={this.state.data}
              margin={{top: 10, right: 0, left: 0, bottom: 30}}>
          <XAxis dataKey="timestamp" ticks={this.getTicks(this.state.data)} tickFormatter={this.dateFormat} />
          <YAxis/>
          <CartesianGrid strokeDasharray="1 1"/>
          <Tooltip labelFormatter={this.toolTipDateFormat}/>
          <Area type='monotone' dataKey='cpu_times_percent_iowait' stroke='#e4c857' fill='#e4c857' />
          <Area type='monotone' dataKey='cpu_times_percent_irq' stroke='#f0e0a0' fill='#f0e0a0' />
          <Area type='monotone' dataKey='cpu_times_percent_ctx_switches' stroke='#8884d8' fill='#8884d8' />
          <Area type='monotone' dataKey='cpu_times_percent_syscalls' stroke='#8884d8' fill='#8884d8' />
          <Area type='monotone' dataKey='cpu_times_percent_interrupts' stroke='#8884d8' fill='#8884d8' />
          <Area type='monotone' dataKey='cpu_times_percent_soft_interrupts' stroke='#8884d8' fill='#8884d8' />
          <Area type='monotone' dataKey='cpu_times_percent_softirq' stroke='#8884d8' fill='#8884d8' />
          <Area type='monotone' dataKey='cpu_times_percent_steal' stroke='#8884d8' fill='#8884d8' />
          <Area type='monotone' dataKey='cpu_times_percent_user' stroke='#f3b318' fill='#f3b318' />
          <Area type='monotone' dataKey='cpu_times_percent_nice' stroke='#8884d8' fill='#8884d8' />
          <Area type='monotone' dataKey='cpu_times_percent_system' stroke='#8884d8' fill='#8884d8' />
          <Area type='monotone' dataKey='cpu_times_percent_idle' stroke='#e4c857' fill='#e4c857' />
          <Area type='monotone' dataKey='cpu_times_percent_guest' stroke='#8884d8' fill='#8884d8' />
          <Area type='monotone' dataKey='cpu_times_percent_guest_nice' stroke='#8884d8' fill='#8884d8' />
        </AreaChart>
      </ResponsiveContainer>
    )
  }
}

//Default properties
CpuStackChart.defaultProps = {
  url: '/fetch/agent/graph/stacked/',
  agentId: '2',
  stackType: 'cpu'
}

//Exports class to Global namespace
export default CpuStackChart;
