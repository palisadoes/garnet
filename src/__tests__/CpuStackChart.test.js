import React from "react";
import ReactDOM from "react-dom";
import { mount, shallow } from "enzyme";
import sinon from "sinon";

import fs from "fs";

import CpuStackChart
  from "../Dashboard/GraphBox/StackedAreaChart/CpuStackChart.js";

describe("CpuStackChart", () => {
  test("should match snapshot", () => {
    const cpuStackChart = shallow(<CpuStackChart />);
    expect(cpuStackChart).toMatchSnapshot();
  });

  test("should fetch data from infoset-ng after mount", () => {
    var route = "/fetch/agent/graphed/stacked",
      agent_Id = "2",
      stack_type = "cpu";
    const url = route + "/" + agent_Id + "/" + stack_type;
    var infoset_dump = __dirname + "/infoset_dump.json";

    var json = JSON.parse(fs.readFileSync(infoset_dump, "utf8"));

    const server = sinon.fakeServer.create();
    server.respondWith("GET", url, (xhr, id, json) => {
      xhr.respond(
        200,
        { "Content-Type": "application/json" },
        JSON.stringify(json)
      );
      const component = mount(<CpuStackChart />);
      server.respond();
      unmount(component);
    });
  });
});
