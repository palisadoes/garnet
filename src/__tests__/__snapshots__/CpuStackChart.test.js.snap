// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CpuStackChart should match snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <ResponsiveContainer
    debounce={0}
    height="35%"
    width="100%"
>
    <AnimationDecorator(ComposedDataDecorator(AreaChart))
        data={Array []}
        height={150}
        layout="horizontal"
        margin={
            Object {
                "bottom": 30,
                "left": 0,
                "right": 0,
                "top": 10,
              }
        }
        stackOffset="none"
        width={600}
    >
        <XAxis
            allowDataOverflow={false}
            allowDecimals={true}
            dataKey="timestamp"
            domain={
                Array [
                    0,
                    "auto",
                  ]
            }
            height={30}
            hide={false}
            mirror={false}
            orientation="bottom"
            padding={
                Object {
                    "left": 0,
                    "right": 0,
                  }
            }
            scale="auto"
            tickCount={5}
            tickFormatter={[Function]}
            ticks={Array []}
            type="category"
            width={0}
            xAxisId={0}
        />
        <YAxis
            allowDataOverflow={false}
            allowDecimals={true}
            domain={
                Array [
                    0,
                    "auto",
                  ]
            }
            height={0}
            hide={false}
            mirror={false}
            orientation="left"
            padding={
                Object {
                    "bottom": 0,
                    "top": 0,
                  }
            }
            scale="auto"
            tickCount={5}
            type="number"
            width={60}
            yAxisId={0}
        />
        <CartesianGrid
            fill="none"
            height={0}
            horizontal={true}
            horizontalPoints={Array []}
            stroke="#ccc"
            strokeDasharray="1 1"
            vertical={true}
            verticalPoints={Array []}
            width={0}
            x={0}
            y={0}
        />
        <Tooltip
            active={false}
            animationDuration={400}
            animationEasing="ease"
            coordinate={
                Object {
                    "x": 0,
                    "y": 0,
                  }
            }
            cursor={true}
            cursorStyle={Object {}}
            isAnimationActive={true}
            itemSorter={[Function]}
            itemStyle={Object {}}
            labelFormatter={[Function]}
            labelStyle={Object {}}
            offset={10}
            separator=" : "
            viewBox={
                Object {
                    "x1": 0,
                    "x2": 0,
                    "y1": 0,
                    "y2": 0,
                  }
            }
            wrapperStyle={Object {}}
        />
        <Area
            activeDot={true}
            animationBegin={0}
            animationDuration={1500}
            animationEasing="ease"
            connectNulls={false}
            curve={true}
            dataKey="cpu_times_percent_softirq"
            dot={false}
            fill="#fd9c04"
            fillOpacity={0.6}
            isAnimationActive={true}
            label={false}
            legendType="line"
            onAnimationEnd={[Function]}
            onAnimationStart={[Function]}
            points={Array []}
            stroke="#fd9c04"
            type="monotone"
            xAxisId={0}
            yAxisId={0}
        />
        <Area
            activeDot={true}
            animationBegin={0}
            animationDuration={1500}
            animationEasing="ease"
            connectNulls={false}
            curve={true}
            dataKey="cpu_times_percent_user"
            dot={false}
            fill="#fdca04"
            fillOpacity={0.6}
            isAnimationActive={true}
            label={false}
            legendType="line"
            onAnimationEnd={[Function]}
            onAnimationStart={[Function]}
            points={Array []}
            stroke="#fdca04"
            type="monotone"
            xAxisId={0}
            yAxisId={0}
        />
        <Area
            activeDot={true}
            animationBegin={0}
            animationDuration={1500}
            animationEasing="ease"
            connectNulls={false}
            curve={true}
            dataKey="cpu_times_percent_nice"
            dot={false}
            fill="#fdca04"
            fillOpacity={0.6}
            isAnimationActive={true}
            label={false}
            legendType="line"
            onAnimationEnd={[Function]}
            onAnimationStart={[Function]}
            points={Array []}
            stroke="#fdca04"
            type="monotone"
            xAxisId={0}
            yAxisId={0}
        />
        <Area
            activeDot={true}
            animationBegin={0}
            animationDuration={1500}
            animationEasing="ease"
            connectNulls={false}
            curve={true}
            dataKey="cpu_times_percent_system"
            dot={false}
            fill="#51D408"
            fillOpacity={0.6}
            isAnimationActive={true}
            label={false}
            legendType="line"
            onAnimationEnd={[Function]}
            onAnimationStart={[Function]}
            points={Array []}
            stroke="#51D408"
            type="monotone"
            xAxisId={0}
            yAxisId={0}
        />
    </AnimationDecorator(ComposedDataDecorator(AreaChart))>
</ResponsiveContainer>,
  "nodes": Array [
    <ResponsiveContainer
      debounce={0}
      height="35%"
      width="100%"
>
      <AnimationDecorator(ComposedDataDecorator(AreaChart))
            data={Array []}
            height={150}
            layout="horizontal"
            margin={
                  Object {
                        "bottom": 30,
                        "left": 0,
                        "right": 0,
                        "top": 10,
                      }
            }
            stackOffset="none"
            width={600}
      >
            <XAxis
                  allowDataOverflow={false}
                  allowDecimals={true}
                  dataKey="timestamp"
                  domain={
                        Array [
                              0,
                              "auto",
                            ]
                  }
                  height={30}
                  hide={false}
                  mirror={false}
                  orientation="bottom"
                  padding={
                        Object {
                              "left": 0,
                              "right": 0,
                            }
                  }
                  scale="auto"
                  tickCount={5}
                  tickFormatter={[Function]}
                  ticks={Array []}
                  type="category"
                  width={0}
                  xAxisId={0}
            />
            <YAxis
                  allowDataOverflow={false}
                  allowDecimals={true}
                  domain={
                        Array [
                              0,
                              "auto",
                            ]
                  }
                  height={0}
                  hide={false}
                  mirror={false}
                  orientation="left"
                  padding={
                        Object {
                              "bottom": 0,
                              "top": 0,
                            }
                  }
                  scale="auto"
                  tickCount={5}
                  type="number"
                  width={60}
                  yAxisId={0}
            />
            <CartesianGrid
                  fill="none"
                  height={0}
                  horizontal={true}
                  horizontalPoints={Array []}
                  stroke="#ccc"
                  strokeDasharray="1 1"
                  vertical={true}
                  verticalPoints={Array []}
                  width={0}
                  x={0}
                  y={0}
            />
            <Tooltip
                  active={false}
                  animationDuration={400}
                  animationEasing="ease"
                  coordinate={
                        Object {
                              "x": 0,
                              "y": 0,
                            }
                  }
                  cursor={true}
                  cursorStyle={Object {}}
                  isAnimationActive={true}
                  itemSorter={[Function]}
                  itemStyle={Object {}}
                  labelFormatter={[Function]}
                  labelStyle={Object {}}
                  offset={10}
                  separator=" : "
                  viewBox={
                        Object {
                              "x1": 0,
                              "x2": 0,
                              "y1": 0,
                              "y2": 0,
                            }
                  }
                  wrapperStyle={Object {}}
            />
            <Area
                  activeDot={true}
                  animationBegin={0}
                  animationDuration={1500}
                  animationEasing="ease"
                  connectNulls={false}
                  curve={true}
                  dataKey="cpu_times_percent_softirq"
                  dot={false}
                  fill="#fd9c04"
                  fillOpacity={0.6}
                  isAnimationActive={true}
                  label={false}
                  legendType="line"
                  onAnimationEnd={[Function]}
                  onAnimationStart={[Function]}
                  points={Array []}
                  stroke="#fd9c04"
                  type="monotone"
                  xAxisId={0}
                  yAxisId={0}
            />
            <Area
                  activeDot={true}
                  animationBegin={0}
                  animationDuration={1500}
                  animationEasing="ease"
                  connectNulls={false}
                  curve={true}
                  dataKey="cpu_times_percent_user"
                  dot={false}
                  fill="#fdca04"
                  fillOpacity={0.6}
                  isAnimationActive={true}
                  label={false}
                  legendType="line"
                  onAnimationEnd={[Function]}
                  onAnimationStart={[Function]}
                  points={Array []}
                  stroke="#fdca04"
                  type="monotone"
                  xAxisId={0}
                  yAxisId={0}
            />
            <Area
                  activeDot={true}
                  animationBegin={0}
                  animationDuration={1500}
                  animationEasing="ease"
                  connectNulls={false}
                  curve={true}
                  dataKey="cpu_times_percent_nice"
                  dot={false}
                  fill="#fdca04"
                  fillOpacity={0.6}
                  isAnimationActive={true}
                  label={false}
                  legendType="line"
                  onAnimationEnd={[Function]}
                  onAnimationStart={[Function]}
                  points={Array []}
                  stroke="#fdca04"
                  type="monotone"
                  xAxisId={0}
                  yAxisId={0}
            />
            <Area
                  activeDot={true}
                  animationBegin={0}
                  animationDuration={1500}
                  animationEasing="ease"
                  connectNulls={false}
                  curve={true}
                  dataKey="cpu_times_percent_system"
                  dot={false}
                  fill="#51D408"
                  fillOpacity={0.6}
                  isAnimationActive={true}
                  label={false}
                  legendType="line"
                  onAnimationEnd={[Function]}
                  onAnimationStart={[Function]}
                  points={Array []}
                  stroke="#51D408"
                  type="monotone"
                  xAxisId={0}
                  yAxisId={0}
            />
      </AnimationDecorator(ComposedDataDecorator(AreaChart))>
</ResponsiveContainer>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <CpuStackChart
        agentId="2"
        stackType="cpu"
        url="/charts/stacked/"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": CpuStackChart {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "agentId": "2",
          "stackType": "cpu",
          "url": "/charts/stacked/",
        },
        "refs": Object {},
        "state": Object {
          "data": Array [],
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <ResponsiveContainer
          debounce={0}
          height="35%"
          width="100%"
>
          <AnimationDecorator(ComposedDataDecorator(AreaChart))
                    data={Array []}
                    height={150}
                    layout="horizontal"
                    margin={
                              Object {
                                        "bottom": 30,
                                        "left": 0,
                                        "right": 0,
                                        "top": 10,
                                      }
                    }
                    stackOffset="none"
                    width={600}
          >
                    <XAxis
                              allowDataOverflow={false}
                              allowDecimals={true}
                              dataKey="timestamp"
                              domain={
                                        Array [
                                                  0,
                                                  "auto",
                                                ]
                              }
                              height={30}
                              hide={false}
                              mirror={false}
                              orientation="bottom"
                              padding={
                                        Object {
                                                  "left": 0,
                                                  "right": 0,
                                                }
                              }
                              scale="auto"
                              tickCount={5}
                              tickFormatter={[Function]}
                              ticks={Array []}
                              type="category"
                              width={0}
                              xAxisId={0}
                    />
                    <YAxis
                              allowDataOverflow={false}
                              allowDecimals={true}
                              domain={
                                        Array [
                                                  0,
                                                  "auto",
                                                ]
                              }
                              height={0}
                              hide={false}
                              mirror={false}
                              orientation="left"
                              padding={
                                        Object {
                                                  "bottom": 0,
                                                  "top": 0,
                                                }
                              }
                              scale="auto"
                              tickCount={5}
                              type="number"
                              width={60}
                              yAxisId={0}
                    />
                    <CartesianGrid
                              fill="none"
                              height={0}
                              horizontal={true}
                              horizontalPoints={Array []}
                              stroke="#ccc"
                              strokeDasharray="1 1"
                              vertical={true}
                              verticalPoints={Array []}
                              width={0}
                              x={0}
                              y={0}
                    />
                    <Tooltip
                              active={false}
                              animationDuration={400}
                              animationEasing="ease"
                              coordinate={
                                        Object {
                                                  "x": 0,
                                                  "y": 0,
                                                }
                              }
                              cursor={true}
                              cursorStyle={Object {}}
                              isAnimationActive={true}
                              itemSorter={[Function]}
                              itemStyle={Object {}}
                              labelFormatter={[Function]}
                              labelStyle={Object {}}
                              offset={10}
                              separator=" : "
                              viewBox={
                                        Object {
                                                  "x1": 0,
                                                  "x2": 0,
                                                  "y1": 0,
                                                  "y2": 0,
                                                }
                              }
                              wrapperStyle={Object {}}
                    />
                    <Area
                              activeDot={true}
                              animationBegin={0}
                              animationDuration={1500}
                              animationEasing="ease"
                              connectNulls={false}
                              curve={true}
                              dataKey="cpu_times_percent_softirq"
                              dot={false}
                              fill="#fd9c04"
                              fillOpacity={0.6}
                              isAnimationActive={true}
                              label={false}
                              legendType="line"
                              onAnimationEnd={[Function]}
                              onAnimationStart={[Function]}
                              points={Array []}
                              stroke="#fd9c04"
                              type="monotone"
                              xAxisId={0}
                              yAxisId={0}
                    />
                    <Area
                              activeDot={true}
                              animationBegin={0}
                              animationDuration={1500}
                              animationEasing="ease"
                              connectNulls={false}
                              curve={true}
                              dataKey="cpu_times_percent_user"
                              dot={false}
                              fill="#fdca04"
                              fillOpacity={0.6}
                              isAnimationActive={true}
                              label={false}
                              legendType="line"
                              onAnimationEnd={[Function]}
                              onAnimationStart={[Function]}
                              points={Array []}
                              stroke="#fdca04"
                              type="monotone"
                              xAxisId={0}
                              yAxisId={0}
                    />
                    <Area
                              activeDot={true}
                              animationBegin={0}
                              animationDuration={1500}
                              animationEasing="ease"
                              connectNulls={false}
                              curve={true}
                              dataKey="cpu_times_percent_nice"
                              dot={false}
                              fill="#fdca04"
                              fillOpacity={0.6}
                              isAnimationActive={true}
                              label={false}
                              legendType="line"
                              onAnimationEnd={[Function]}
                              onAnimationStart={[Function]}
                              points={Array []}
                              stroke="#fdca04"
                              type="monotone"
                              xAxisId={0}
                              yAxisId={0}
                    />
                    <Area
                              activeDot={true}
                              animationBegin={0}
                              animationDuration={1500}
                              animationEasing="ease"
                              connectNulls={false}
                              curve={true}
                              dataKey="cpu_times_percent_system"
                              dot={false}
                              fill="#51D408"
                              fillOpacity={0.6}
                              isAnimationActive={true}
                              label={false}
                              legendType="line"
                              onAnimationEnd={[Function]}
                              onAnimationStart={[Function]}
                              points={Array []}
                              stroke="#51D408"
                              type="monotone"
                              xAxisId={0}
                              yAxisId={0}
                    />
          </AnimationDecorator(ComposedDataDecorator(AreaChart))>
</ResponsiveContainer>,
        "_debugID": 2,
        "_renderedOutput": <ResponsiveContainer
          debounce={0}
          height="35%"
          width="100%"
>
          <AnimationDecorator(ComposedDataDecorator(AreaChart))
                    data={Array []}
                    height={150}
                    layout="horizontal"
                    margin={
                              Object {
                                        "bottom": 30,
                                        "left": 0,
                                        "right": 0,
                                        "top": 10,
                                      }
                    }
                    stackOffset="none"
                    width={600}
          >
                    <XAxis
                              allowDataOverflow={false}
                              allowDecimals={true}
                              dataKey="timestamp"
                              domain={
                                        Array [
                                                  0,
                                                  "auto",
                                                ]
                              }
                              height={30}
                              hide={false}
                              mirror={false}
                              orientation="bottom"
                              padding={
                                        Object {
                                                  "left": 0,
                                                  "right": 0,
                                                }
                              }
                              scale="auto"
                              tickCount={5}
                              tickFormatter={[Function]}
                              ticks={Array []}
                              type="category"
                              width={0}
                              xAxisId={0}
                    />
                    <YAxis
                              allowDataOverflow={false}
                              allowDecimals={true}
                              domain={
                                        Array [
                                                  0,
                                                  "auto",
                                                ]
                              }
                              height={0}
                              hide={false}
                              mirror={false}
                              orientation="left"
                              padding={
                                        Object {
                                                  "bottom": 0,
                                                  "top": 0,
                                                }
                              }
                              scale="auto"
                              tickCount={5}
                              type="number"
                              width={60}
                              yAxisId={0}
                    />
                    <CartesianGrid
                              fill="none"
                              height={0}
                              horizontal={true}
                              horizontalPoints={Array []}
                              stroke="#ccc"
                              strokeDasharray="1 1"
                              vertical={true}
                              verticalPoints={Array []}
                              width={0}
                              x={0}
                              y={0}
                    />
                    <Tooltip
                              active={false}
                              animationDuration={400}
                              animationEasing="ease"
                              coordinate={
                                        Object {
                                                  "x": 0,
                                                  "y": 0,
                                                }
                              }
                              cursor={true}
                              cursorStyle={Object {}}
                              isAnimationActive={true}
                              itemSorter={[Function]}
                              itemStyle={Object {}}
                              labelFormatter={[Function]}
                              labelStyle={Object {}}
                              offset={10}
                              separator=" : "
                              viewBox={
                                        Object {
                                                  "x1": 0,
                                                  "x2": 0,
                                                  "y1": 0,
                                                  "y2": 0,
                                                }
                              }
                              wrapperStyle={Object {}}
                    />
                    <Area
                              activeDot={true}
                              animationBegin={0}
                              animationDuration={1500}
                              animationEasing="ease"
                              connectNulls={false}
                              curve={true}
                              dataKey="cpu_times_percent_softirq"
                              dot={false}
                              fill="#fd9c04"
                              fillOpacity={0.6}
                              isAnimationActive={true}
                              label={false}
                              legendType="line"
                              onAnimationEnd={[Function]}
                              onAnimationStart={[Function]}
                              points={Array []}
                              stroke="#fd9c04"
                              type="monotone"
                              xAxisId={0}
                              yAxisId={0}
                    />
                    <Area
                              activeDot={true}
                              animationBegin={0}
                              animationDuration={1500}
                              animationEasing="ease"
                              connectNulls={false}
                              curve={true}
                              dataKey="cpu_times_percent_user"
                              dot={false}
                              fill="#fdca04"
                              fillOpacity={0.6}
                              isAnimationActive={true}
                              label={false}
                              legendType="line"
                              onAnimationEnd={[Function]}
                              onAnimationStart={[Function]}
                              points={Array []}
                              stroke="#fdca04"
                              type="monotone"
                              xAxisId={0}
                              yAxisId={0}
                    />
                    <Area
                              activeDot={true}
                              animationBegin={0}
                              animationDuration={1500}
                              animationEasing="ease"
                              connectNulls={false}
                              curve={true}
                              dataKey="cpu_times_percent_nice"
                              dot={false}
                              fill="#fdca04"
                              fillOpacity={0.6}
                              isAnimationActive={true}
                              label={false}
                              legendType="line"
                              onAnimationEnd={[Function]}
                              onAnimationStart={[Function]}
                              points={Array []}
                              stroke="#fdca04"
                              type="monotone"
                              xAxisId={0}
                              yAxisId={0}
                    />
                    <Area
                              activeDot={true}
                              animationBegin={0}
                              animationDuration={1500}
                              animationEasing="ease"
                              connectNulls={false}
                              curve={true}
                              dataKey="cpu_times_percent_system"
                              dot={false}
                              fill="#51D408"
                              fillOpacity={0.6}
                              isAnimationActive={true}
                              label={false}
                              legendType="line"
                              onAnimationEnd={[Function]}
                              onAnimationStart={[Function]}
                              points={Array []}
                              stroke="#51D408"
                              type="monotone"
                              xAxisId={0}
                              yAxisId={0}
                    />
          </AnimationDecorator(ComposedDataDecorator(AreaChart))>
</ResponsiveContainer>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <CpuStackChart
    agentId="2"
    stackType="cpu"
    url="/charts/stacked/"
/>,
}
`;
